opm/simulators/utils/readDeck.cpp

```diff
@@ -228,9 +229,13 @@ void readDeck(int rank, std::string& deckFilename, std::unique_ptr<Opm::Deck>& d
 
             Opm::checkConsistentArrayDimensions(*eclipseState, *schedule, *parseContext, *errorGuard);
         }
-        catch(const std::exception& e)
+        catch(const OpmInputError& input_error) {
+            failureMessage = input_error.what();
+            parseSuccess = 0;
+        }
+        catch(const std::exception& std_error)
         {
-            failureMessage = e.what();
+            failureMessage = std_error.what();
```

Comment:
I think this is an unexpected error (maybe programming error), but failureMessage will never be printed.

Reply:
Yes - currently you are right; the `Deck`is created like this:
```
try {
     // parse file; might fail with either std::exception or OpmInputError
} catch (const OpmInputError& opm_error) {
     throw;
} catch (const std::exception& std_error) {
     std::string msg_fmt = fmt::format("Problem with keyword {{keyword}}\n"
                                       "In {{file}} line {{line}}\n",
                                       "Internal error message: {}" , std_error.what());
     OpmLog::error(OpmInputError::format( msg_fmt, keyword.location() );
     throw;
}
``` 
this construction "guarantees" that the error message from the `std::exception` has been printed - with accompanying location information. The intention is to add a similar construction for the construction of `EclipseState, Schedule` and `SummaryConfig` - but you are right; before that has been done we will not see the messages from the construction of `EclipseState, Schedule` and `SummaryConfig`.

I will create a opm-common PR ensuring that.